tools/trace2syz/trace2syz/intermediateTypes.go

```diff
@@ -0,0 +1,686 @@
+package trace2syz
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/google/syzkaller/prog"
+	"strconv"
+)
+
+type Operation int
+
+const (
+	ORop       = iota //OR = |
+	ANDop             //AND = &
+	XORop             //XOR = ^
+	NOTop             //NOT = !
+	LSHIFTop          //LSHIFT = <<
+	RSHIFTop          //RSHIFT = >>
+	ONESCOMPop        //ONESCOMP = ~
+	TIMESop           //TIMES = *
+	LANDop            //LAND = &&
+	LORop             //LOR = ||
+	LEQUALop          //LEQUAL = ==
+)
+
+//TraceTree struct contains intermediate representation of trace
+//If a trace is multiprocess it constructs a trace for each type
+type TraceTree struct {
+	TraceMap map[int64]*Trace
+	Ptree    map[int64][]int64
+	RootPid  int64
+	Filename string
+}
+
+//NewTraceTree initializes a TraceTree
+func NewTraceTree() (tree *TraceTree) {
+	tree = &TraceTree{
+		TraceMap: make(map[int64]*Trace),
+		Ptree:    make(map[int64][]int64),
+		RootPid:  -1,
+	}
+	return
+}
+
+func (tree *TraceTree) Contains(pid int64) bool {
+	if _, ok := tree.TraceMap[pid]; ok {
+		return true
+	}
+	return false
+}
+
+func (tree *TraceTree) Add(call *Syscall) *Syscall {
+	if tree.RootPid < 0 {
+		tree.RootPid = call.Pid
+	}
+	if !call.Resumed {
+		if !tree.Contains(call.Pid) {
+			tree.TraceMap[call.Pid] = newTrace()
+			tree.Ptree[call.Pid] = make([]int64, 0)
+		}
+	}
+	c := tree.TraceMap[call.Pid].Add(call)
+	if c.CallName == "clone" && !c.Paused {
+		tree.Ptree[c.Pid] = append(tree.Ptree[c.Pid], c.Ret)
+	}
+	return c
+}
+
+func (tree *TraceTree) String() string {
+	var buf bytes.Buffer
+
+	buf.WriteString(fmt.Sprintf("Root: %d\n", tree.RootPid))
+	buf.WriteString(fmt.Sprintf("Pids: %d\n", len(tree.TraceMap)))
+	return buf.String()
+}
+
+//Trace is just a list of system calls
+type Trace struct {
+	Calls []*Syscall
+}
+
+//newTrace initializes a new trace
+func newTrace() (trace *Trace) {
+	trace = &Trace{Calls: make([]*Syscall, 0)}
+	return
+}
+
+func (trace *Trace) Add(call *Syscall) (ret *Syscall) {
+	if call.Resumed {
+		lastCall := trace.Calls[len(trace.Calls)-1]
+		lastCall.Args = append(lastCall.Args, call.Args...)
+		lastCall.Paused = false
+		lastCall.Ret = call.Ret
+		ret = lastCall
+	} else {
+		trace.Calls = append(trace.Calls, call)
+		ret = call
+	}
+	return
+}
+
+//Syscall struct is the IR type for any system call
+type Syscall struct {
+	CallName string
+	Args     []irType
+	Pid      int64
+	Ret      int64
+	Cover    []uint64
+	Paused   bool
+	Resumed  bool
+}
+
+//NewSyscall - constructor
+func NewSyscall(pid int64, name string, args []irType, ret int64, paused bool, resumed bool) (sys *Syscall) {
```

Comment:
It seems that pid is always -1 in all callers, so probably just use -1 here.
change `paused bool, resumed bool` to `paused, resumed bool`

Reply:
pid can definitely be > -1.  In the strace.y and generated code it is called with positive values.